Zend/zend_API.h

```diff
@@ -1972,6 +2016,30 @@ static zend_always_inline int zend_parse_arg_class_name_or_obj(
 	return 1;
 }
 
+static zend_always_inline int zend_parse_arg_str_or_obj(
+	zval *arg, zend_string **destination_string, zend_object **destination_object, zend_class_entry *base_ce, int allow_null
+) {
+	if (EXPECTED(Z_TYPE_P(arg) == IS_STRING)) {
+		*destination_string = Z_STR_P(arg);
+		*destination_object = NULL;
+	} else if (EXPECTED(Z_TYPE_P(arg) == IS_OBJECT)) {
```

Comment:
I'd move the object case first. I believe in places where it's used it's the "more expected" one.